homeassistant/components/vesync/light.py

```diff
@@ -175,3 +180,165 @@ def color_temp(self) -> int:
+        """Return the current color mode."""
+        # get value from pyvesync library api,
+        color_mode_value = self.device.color_mode
+        if color_mode_value.lower() == "white":
+            return ColorMode.COLOR_TEMP
+        if color_mode_value.lower() in ["hsv", "color"]:
+            return ColorMode.HS
+        return ColorMode.ONOFF
+
+    @property
+    def hs_color(self) -> tuple[float, float] | None:
+        """Return the current color mode."""
+        # get value from pyvesync library api,
+        color_value_hsv = None
+        try:
+            # get hsv named tuple from pyvesync library (hue, sat, value)
+            color_value_hsv = self.device.color_value_hsv
+        except ValueError:
+            # deal if any unexpected/non numeric value
+            _LOGGER.debug(
+                "VeSync - received unexpected 'color_value_hsv' value "
+                "from pyvesync api: %s",
+                str(color_value_hsv),
+            )
+            return None
+        hs_color = color_value_hsv.hue, color_value_hsv.saturation
+        return hs_color
+
+    @property
+    def brightness(self) -> int | None:
+        """Get light brightness."""
+        # If Color Mode = COLOR use HSV value
+        if self.color_mode == ColorMode.HS:
+            # get value from pyvesync library api,
+            color_value_hsv = None
+            try:
+                # get hsv named tuple from pyvesync library (hue, sat, value)
+                color_value_hsv = self.device.color_value_hsv
+                # check for validity of brightness value received
+                hsv_value = int(color_value_hsv.value)
+            except ValueError:
+                # deal if any unexpected/non numeric value
+                _LOGGER.debug(
+                    "VeSync - received unexpected 'color_value_hsv' value "
+                    "from pyvesync api: %s",
+                    str(color_value_hsv),
+                )
+                return 0
+            # convert percent brightness to ha expected range
+            return round((max(1, hsv_value) / 100) * 255)
+
+        # get value from pyvesync library api,
+        result = self.device.brightness
+        try:
+            # check for validity of brightness value received
+            brightness_value = int(result)
+        except ValueError:
+            # deal if any unexpected/non numeric value
+            _LOGGER.debug(
+                "VeSync - received unexpected 'brightness' value "
+                "from pyvesync api: %s",
+                str(result),
+            )
+            return 0
+        # convert percent brightness to ha expected range
+        return round((max(1, brightness_value) / 100) * 255)
+
+    def turn_on(self, **kwargs) -> None:
+        """Turn the device on, and adjust attributes."""
+        # The Valceno Multicolor bulb use the V2 API endpoint,
+        #   so it has the capability of setting every light parameter in a single call
```

Comment:
Hey, thank you so much for review it!
At the time of writing this PR, the valceno bulbs were the only multicolor bulb in existence on the pyvesync library, since I wrote the support for it, in conjunction of this PR.

Since it took long enough for this pr to be reviewed and merged. The pyvesync library already got support for a second multicolor bulb, which uses a different API endpoint internally, and RGB values instead of HSV. but that is out of scope of this PR.
It makes no sense to make edits to this PR to add yet another bulb model class with different properties. at this time. The PR would get too big, hard to review, and more unrelated changes.

I want to move over with this pr merged at once, because I DO want to help implement the other multicolor bulb as well, but one thing at time.

There is no risk of the multicolor_bulb class being wrongly used by a different model at this time, because the other models are not registered on this integration yet. That would be done in a next step, but then the classes can be adjusted accordingly, giving them proper support.

> is it possible that there needs to be additional checks to ensure that future multicolor bulb additions uses a particular version of the api?

The home assistant component side of it doesn't do any api call. It only interacts with the pyvesync library. So the code to handle different versions of the api was already baked on the library. The homeassistant component should have no problem with that.

I've been using the code on this PR daily for a year now. It's tried and true at this point. Other than occasional server issues which are unrelated. even the native app timeouts sometimes.

Reply:
Hey, thank you so much for review it!
At the time of writing this PR, the valceno bulbs were the only multicolor bulb in existence on the pyvesync library, since I wrote the support for it, in conjunction of this PR.

Since it took long enough for this pr to be reviewed and merged. The pyvesync library already got support for a second multicolor bulb, which uses a different API endpoint internally, and RGB values instead of HSV. but that is out of scope of this PR.
It makes no sense to make edits to this PR to add yet another bulb model class with different properties. at this time. The PR would get too big, hard to review, and more unrelated changes.

I want to move over with this pr merged at once, because I DO want to help implement the other multicolor bulb as well, but one thing at time.

There is no risk of the multicolor_bulb class being wrongly used by a different model at this time, because the other models are not registered on this integration yet. That would be done in a next step, but then the classes can be adjusted accordingly, giving them proper support.

> is it possible that there needs to be additional checks to ensure that future multicolor bulb additions uses a particular version of the api?

The home assistant component side of it doesn't do any api call. It only interacts with the pyvesync library. So the code to handle different versions of the api was already baked on the library. The homeassistant component should have no problem with that.

I've been using the code on this PR daily for a year now. It's tried and true at this point. Other than occasional server issues which are unrelated. even the native app timeouts sometimes.